records/state/state.go

```diff
@@ -123,12 +131,22 @@ var sources = map[string]func(*Task) []string{
 func hostIPs(t *Task) []string { return []string{t.SlaveIP} }
 
 // networkInfoIPs returns IP addresses from a given Task's
-// []Status.ContainerStatus.[]NetworkInfos.IPAddress
+// []Status.ContainerStatus.[]NetworkInfos.[]IPAddresses.IPAddress
 func networkInfoIPs(t *Task) []string {
 	return statusIPs(t.Statuses, func(s *Status) []string {
 		ips := make([]string, len(s.ContainerStatus.NetworkInfos))
-		for i, netinfo := range s.ContainerStatus.NetworkInfos {
-			ips[i] = netinfo.IPAddress
+		for _, netinfo := range s.ContainerStatus.NetworkInfos {
+			if len(netinfo.IPAddresses) > 0 {
```

Comment:
I'll defer to your judgement here; my instinct is to code defensively, but
I am not very familiar with the broader mesosphere ecosystem (i.e. what
could be generating the state.json) so if you think it's safe to simplify
then I'm happy to do so.

On Wed, Nov 18, 2015 at 10:27 AM, James DeFelice <notifications@github.com>
wrote:

> In records/state/state.go
> <https://github.com/mesosphere/mesos-dns/pull/355#discussion_r45238728>:
>
> >  func networkInfoIPs(t *Task) []string {
> >  	return statusIPs(t.Statuses, func(s *Status) []string {
> >  		ips := make([]string, len(s.ContainerStatus.NetworkInfos))
> > -		for i, netinfo := range s.ContainerStatus.NetworkInfos {
> > -			ips[i] = netinfo.IPAddress
> > +		for _, netinfo := range s.ContainerStatus.NetworkInfos {
> > +			if len(netinfo.IPAddresses) > 0 {
>
> understood. i was under the impression that no version of mesos would ever
> generate network-info that would contain both.
>
> that said, this is really a minor comment - intent was to further simplify
> the code. if we're really worried about a botched mesos impl then we can
> keep this as is
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/355/files#r45238728>.
>